internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -194,6 +194,28 @@ func (r *BMACReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (c
 		return reconcileComplete{}.Result()
 	}
 
+	// Should we check the status of the BMH resource?
+	// Provisioning / Deprovisioning ?
+	// What happens if we do the reconcile on a BMH that
+	// is in a Deprovisioning state?
+
+	agent := r.findAgent(ctx, bmh)
```

Comment:
Let's get the PR in as-is but I think we should, eventually, improve the flow. One nice thing of doing the BMH reconcile first is that the `findAgent` function is not super cheap and it's not needed unless the detached annotation is not set.